client/rest-high-level/src/test/java/org/elasticsearch/client/documentation/CRUDDocumentationIT.java

```diff
@@ -497,9 +493,34 @@ public void onFailure(Exception e) {
             request.waitForActiveShards(ActiveShardCount.ALL); // <2>
             // end::update-request-active-shards
         }
+        {
+            UpdateRequest request = new UpdateRequest("posts", "doc", "async").doc("reason", "async update").docAsUpsert(true);
+
+            // tag::update-execute-async
+            client.updateAsync(request, new ActionListener<UpdateResponse>() {
+                @Override
+                public void onResponse(UpdateResponse updateResponse) {
+                    // <1>
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    // <2>
+                }
+            });
+            // end::update-execute-async
+
+            assertTrue(awaitBusy(() -> {
+                try {
+                    return client.exists(new GetRequest("posts", "doc", "async"));
+                } catch (IOException e) {
+                    return false;
+                }
+            }));
```

Comment:
once you have changed the id of the update, it won't sneak in anymore and cause version conflict, so I think it doesn't necessarily needs to be moved down? But maybe it's good like you did. Maybe we don't even need to wait for it at this point?

Reply:
That was for readability, as some test use the same document to index or update multiple times. But I can move them back if you prefer.

> Maybe we don't even need to wait for it at this point?

We need to wait  for any writing operation, but we can relax this for read operations.